### DISCUSSION TITLE
Product variants - the main value of the quantity of products

### DISCUSSION THREAD

**Discussion by gs2k84** [PrestaShop/PrestaShop#40173] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Describe the solution you'd like

Hello,

I believe that the function of one main product quantity in stock for product variants is badly needed.
You can choose whether you want the product variants to have their own quantity management in stock or there is one main value for all variants.

Example.

My company sells engines for cars.

You can buy the engine in option:
1.engine without accessories
2. engine with fuel system
3. engine with complete accessories

At this point, for each variant, I have to set a separate product condition 1 piece.

However, it is still the same engine. So I have 1 piece of this engine in stock, but it gives the customer the opportunity to buy it in various options.

When he buys variant 1, the engine without accessories, the other two are still in stock (according to prestashop), they are physically gone because the engine has been sold. Only the accessories for this engine remain, which will be displayed as separate advertisements after disassembling it from the engine.

Therefore, being able to set the main quantity to one piece would solve the problem.

**Comment by hibatallahAouadni** [comment]
Hello @gs2k84 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you
